plugins/pulp_rpm/plugins/distributors/yum/metadata/repomd.py

```diff
@@ -107,8 +108,7 @@ def add_metadata_file_metadata(self, data_type, file_path, precalculated_checksu
         # calculating it again.
         if precalculated_checksum is None:
             with open(file_path, 'rb') as file_handle:
-                content = file_handle.read()
-                checksum_element.text = self.checksum_constructor(content).hexdigest()
+                checksum_element.text = self._calculate_checksum(file_handle)
```

Comment:
Minor style nit, the leading underscore is used to signify "private" but it's being treated as a public one.

Reply:
Just a bit lazy to use public method here because I wanted to avoid adding comment docs to explain the use of each param. :) Let me know if you want me to remove the leading underscore.

Reply:
Go ahead and it, but I won't complain if the comment docs are underwhelming

Reply:
done